logistics_project/apps/malawi/warehouse/report_views/stock_status.py

```diff
@@ -13,133 +13,157 @@
 from django.db.models.aggregates import Sum
 from django.shortcuts import get_object_or_404
 
+
 class View(warehouse_view.DistrictOnlyView):
 
-    def custom_context(self, request):        
+    def _get_product_status_table(self, supply_points, product, date):
+        ret = []
+        for s in supply_points:
+            qs = ProductAvailabilityData.objects.filter(supply_point=s, product=product, date=date)
```

Comment:
my general approach has been to not include the is_facility flag in the filter when we query for a specific product, since we will have already applied the filter to the product that gets passed in here. this reduces redundancy and even if we put the filter it wouldn't fix the issue if you're still passing the wrong product in (we'd just end up with 0s)